xonsh/completer.py

```diff
@@ -157,20 +199,19 @@ def complete(self, prefix, line, begidx, endidx, ctx=None):
                     s = s.rstrip() + slash
                 rtn.add(s)
             if len(rtn) == 0:
-                rtn = self.path_complete(prefix)
-            return sorted(rtn)
+                rtn = self.path_complete(prefix, path_str_start, path_str_end)
         elif prefix.startswith('${') or prefix.startswith('@('):
             # python mode explicitly
             rtn = set()
         elif prefix.startswith('-'):
-            return sorted(self._man_completer.option_complete(prefix, cmd))
+            return lprefix, sorted(self._man_completer.option_complete(prefix, cmd))
```

Comment:
Ok so it seems that lprefix is just used by prompt-toolkit. Also it seems like lprefix should come after the completions, to be consistent with  prompt-toolkit, but it is your call.